client/components/inline-support-link/index.jsx

```diff
@@ -75,7 +76,7 @@ class InlineSupportLink extends Component {
 			<LinkComponent
 				className="inline-support-link"
 				href={ url }
-				onClick={ openDialog }
+				onClick={ shouldOpenDialog && openDialog }
```

Comment:
Looks like my suggestion for this was not very good as it's getting a warning now that `onClick` is expecting a listener function or undefined and it's received `false`. So this should be either `shouldOpenDialog ? openDialog : undefined` or moved back within the `openDialog` handler as it was before.